ui-modules/app-inspector/app/components/stream/stream.directive.js

```diff
@@ -43,19 +43,51 @@ export function streamDirective() {
         $scope.autoUpdate = $scope.autoUpdate !== false;
         $scope.tail = $scope.tail !== false;
 
-        let pre = $element.find('pre');
+        // Content filtering features
+        $scope.filteredStream = [];
+        $scope.streamProcessedUpTo = 0;
+        $scope.isDisplayOther = $scope.isDisplayOther !== false;
+        $scope.isDisplayError = $scope.isDisplayError !== false;
+        $scope.isDisplayDebug = $scope.isDisplayDebug !== false;
+        $scope.isDisplayTrace = $scope.isDisplayTrace !== false;
+        $scope.isDisplayWarning = $scope.isDisplayWarning !== false;
+        $scope.isFilterContent = isFilterContent;
+        $scope.isDisplayFormattedItem = isDisplayFormattedItem;
+        $scope.getFormattedItemLogLevel = getFormattedItemLogLevel;
+
+        // CLI XML features
+        $scope.cliXml = false;
+        $scope.cliXmlIdentified = false;
+        $scope.toggleCliXml = toggleCliXml;
+        $scope.isCliXmlSupported = isCliXmlSupported;
+        $scope.cliXmlVerificationRequired = isWinRmStream(); // CLI XML verification is required only when stream is WinRM
+
+        let autoScrollableElement = Array.from(document.getElementsByClassName('auto-scrollable'));
```

Comment:
I think we need to restrict to things in the current `$element` -- otherwise an `auto-scrollable` anywhere in the DOM will end up changing the `$scope` of this, no?

Reply:
Agree, improvement in the new commit.